src/error.rs

```diff
@@ -25,6 +25,18 @@ pub enum DesignSpaceLoadError {
     DeError(#[from] DeError),
 }
 
+/// An error that occurs while attempting to write a designspace file to disk.
+#[derive(Debug, Error)]
+pub enum DesignSpaceSaveError {
+    /// An [`std::io::Error`].
+    #[error("failed to open/write file")]
```

Comment:
It's useful to also print the inner error here, which may tell us a bit more
```suggestion
    #[error("failed to load designspace: '{0}'")]
```